AsyncDisplayKit/ASCellNode.h

```diff
@@ -14,6 +14,15 @@ NS_ASSUME_NONNULL_BEGIN
 
 typedef NSUInteger ASCellNodeAnimation;
 
+typedef enum : NSUInteger {
+  // Indicates a cell has just became visible
+  ASCellNodeVisibilityEventVisible,
+  // Indicates a visible cell has a different visible rect. Its position, percent visible or both have changed.
```

Comment:
"Its position (determined by scrollView.contentOffset) has changed while at least 1px remains visible.  It is possible that 100% of the cell is visible both before and after and only its position has changed, or that the position change has resulted in more or less of the cell being visible.  Use CGRectIntersect between cellFrame and scrollView.bounds to get this rectangle"